### DISCUSSION TITLE
Spatial Mesh Observer in MRTK3

### DISCUSSION THREAD

**Discussion by Ram1508** [microsoft/MixedRealityToolkit-Unity#11531] [Q&A] [answered]
## Describe the issue
Hello, I was working on Spatial Mesh concept in  MRTK3 and learnt that Spatial Awareness System produced by MRTKv2 services are now replaced by ARMeshManager which consumes XRmeshSystem. MRTKv2 has spatialAwarnessSystem that provides " ResumeObservers()" to to start/restart and "SuspendObservers()" to stop/pause spatial observer.
MRTK3 uses XrMeshSystem which has "GenerateMeshAsync" to generate mesh and the ARMeshManager which consuming XRMeshSystem has "DestroyAllMeshes()" to destroy all the generated mesh.
But is there any functionality same as MRTKv2 for to start or stop the observer. Is there anything I am missing? 

## Existing doc link
https://learn.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/mrtk2/features/spatial-awareness/spatial-awareness-getting-started?view=mrtkunity-2022-05

## Additional context
I am using Unity 2021.3.16.f1, Mixed Reality OpenXR Plugin 1.8.0,  MRTK3 pre.14

 Can anyone guide me?

**Comment by AMollis** [comment]
Hi @Ram1508 , thank you for the question.   I've converted your issue to a discussion topic, and will be tracking down an answer for you.

**Reply by AMollis** [reply]
@Ram1508 , disabling the `ARMeshManager` will call stop on the `XRMeshSubsystem`.  Code from `ARMeshManager.cs`:

```
        void OnDisable()
        {
            if (m_Subsystem != null && m_Subsystem.running)
                m_Subsystem.Stop();
        }
```

Is this what you are looking for?

**Comment by Ram1508** [comment]
@AMollis thanks for the reply, disabling the ARMeshManager stops the XRMeshSubsystem. But I need to change the mesh material w.r.t the display option  like visible/occlusive based upon the subsytem status. It is provided by SpatialAwarenessMeshDisplayOptions in MRTKv2. 

MRTK3:
ARMeshManager provides access to created meshes. 
![Screenshot (26)](https://user-images.githubusercontent.com/131968892/236825553-0c3951d8-0f65-4f63-9c01-c81b29205d39.png)
I changed the mesh material based on the use case.

**Reply by AMollis** [reply, marked as answer]
The MRTK3 doesn't provide any scripts to help here.  I think what you are doing is the right approach for MRTK3.

**Reply by AMollis** [reply]
You could also try editing the mesh manager's mesh prefab and recreate the mesh object via `meshManager.DestroyAllMeshes()`. But I haven't tried this out.
